src/couch/src/couch_httpd_auth.erl

```diff
@@ -460,16 +460,17 @@ cookie_auth_header(
 cookie_auth_header(_Req, _Headers) ->
     [].
 
-cookie_auth_cookie(Req, User, Secret, TimeStamp, MustMatchBasic) ->
-    MustMatchBasicStr =
-        case MustMatchBasic of
-            true -> "1";
-            false -> "0"
-        end,
+cookie_auth_cookie(Req, User, Secret, TimeStamp, true) ->
     SessionData = lists:join(":", [
         User,
-        lists:join(",", [erlang:integer_to_list(TimeStamp, 16), MustMatchBasicStr])
+        lists:join(",", [erlang:integer_to_list(TimeStamp, 16), "1"])
     ]),
+    cookie_auth_cookie(Req, Secret, SessionData);
+cookie_auth_cookie(Req, User, Secret, TimeStamp, false) ->
+    SessionData = ?b2l(User) ++ ":" ++ erlang:integer_to_list(TimeStamp, 16),
```

Comment:
Why do we convert User to `?b2l(User)` while the previous one just used `User`?

Reply:
because the earlier PR removed some unnecessary b2l/l2b code, and this partially reverses it.

but now I look at I think I prefer `SessionData = lists:join(":", [User, erlang:integer_to_list(TimeStamp, 16)]),` for the old cookie generation. cleaner that way and matches the new clause.